hydromt/validators/data_catalog.py

```diff
@@ -0,0 +1,163 @@
+
+    time_range: TimeRange
+    bbox: Bbox
+
+
+class DataCatalogMetaData(BaseModel):
+    """The metadata section of a Hydromt data catalog."""
+
+    root: Optional[Path] = None
+    version: Optional[Union[str, int]] = None
+    name: Optional[str] = None
+    model_config: ConfigDict = ConfigDict(
+        str_strip_whitespace=True,
+        extra="allow",
+    )
+
+    @staticmethod
+    def from_dict(input_dict: Optional[Dict]) -> "DataCatalogMetaData":
+        """Convert a dictionary into a validated data catalog metadata item."""
+        if input_dict is None:
+            return DataCatalogMetaData()
+        else:
+            return DataCatalogMetaData(**input_dict)
+
+
+class DataCatalogItemMetadata(BaseModel):
+    """The metadata for a data source."""
+
+    category: Optional[str] = None
+    paper_doi: Optional[str] = None
+    paper_ref: Optional[str] = None
+    source_license: Optional[str] = None
+    source_url: Optional[AnyUrl] = None
+    source_version: Optional[str] = None
+    notes: Optional[str] = None
+
+    model_config: ConfigDict = ConfigDict(
+        str_strip_whitespace=True, coerce_numbers_to_str=True
+    )
+
+    @staticmethod
+    def from_dict(input_dict):
+        """Convert a dictionary into a validated source metadata item."""
+        if input_dict is None:
+            return DataCatalogItemMetadata()
+        else:
+            item_source_url = input_dict.pop("source_url", None)
+            if item_source_url:
+                Url(item_source_url)
+            return DataCatalogItemMetadata(**input_dict, source_url=item_source_url)
+
+
+class DataCatalogItem(BaseModel):
+    """A validated data source."""
+
+    name: str
+    data_type: str
+    driver: str
+    path: Path
+    crs: Optional[int] = None
+    filesystem: Optional[str] = None
+    kwargs: Dict[str, Any] = {}
+    storage_options: Dict[str, Any] = {}
+    rename: Dict[str, str] = {}
+    nodata: Optional[Number] = None
+    meta: Optional[DataCatalogItemMetadata] = None
+    unit_add: Optional[Dict[str, Number]] = None
+    unit_mult: Optional[Dict[str, Number]] = None
+
+    model_config: ConfigDict = ConfigDict(
+        str_strip_whitespace=True,
+        extra="forbid",
+    )
+
+    @staticmethod
+    def from_dict(input_dict, name=None):
+        """Convert a dictionary into a validated source item."""
+        dict_name = input_dict.pop("name", None)
+        if name is None:
+            entry_name = dict_name
+        else:
+            entry_name = name
+        item_metadata = DataCatalogItemMetadata.from_dict(input_dict.pop("meta", {}))
+        item_kwargs = input_dict.pop("kwargs", {})
+        item_storage_options = input_dict.pop("storage_options", {})
+        return DataCatalogItem(
+            **input_dict,
+            name=entry_name,
+            kwargs=item_kwargs,
+            storage_options=item_storage_options,
+            meta=item_metadata,
+        )
```

Comment:
This is an alternative for the built-in `DataCatalogItem.model_validate` static method (parses python dict and returns an instance of the model). Do you want to overwrite this method or is this an alternative?

Reply:
It might be interesting to explore this in the future, but for now, I'll stick with the current approach. Good comment!